numba_dpex/core/kernel_interface/spirv_kernel.py

```diff
@@ -139,6 +139,20 @@ def compile(
         self._llvm_module = kernel.module.__str__()
         self._module_name = kernel.name
 
+        # Dump LLVM IR if DEBUG flag is set.
+        if config.DEBUG:
+            import hashlib
+
+            # Embed hash of module name in the output file name
+            # so that different kernels are written to separate files
+            with open(
+                "llvm_kernel_"
```

Comment:
We can use `self._module_name` in the file name so that it is clear. Although, with the type of kernel names we generate for parfors that might be exposing skeletons on the closet 😄

Reply:
:speak_no_evil: